Automattic/wp-calypso
# Plans 2023: Fix 3Y plan price descriptions

Fixes https://github.com/Automattic/wp-calypso/issues/74638

## Proposed Changes

- Updates the 3Y plan price descriptions to show the correct text.
- Adds `currencyCode` to the data store product (and uses that in the respective component, in place of the Calypso selector)

### Media

**Before**

<img width="1265" alt="227237260-185a84d0-4630-4d6d-a32a-f666c368eca1" src="https://user-images.githubusercontent.com/1705499/229797016-03d654b0-e21d-4188-b953-f0c840e9da73.png">

**After**

<img width="1273" alt="227233544-91efa92e-c0ce-422d-86e8-a641845836a4" src="https://user-images.githubusercontent.com/1705499/229796987-53ea2876-fe15-47bb-9452-8444d3e278a7.png">

### TODO

- [x] Ensure the selectors/resolvers are called at the right place to avoid any flicker with the store data coming in after the grid is rendered

## Testing Instructions



* Go to `/plans/3yearly/[SITE]` and ensure the price descriptions show the prices, as in the media. Do the same for `/plans/yearly` and `/plans/2yearly`.
* Go to `/start/plans` and ensure things render as before.
* In all cases above, there should be no flicker of price descriptions changing. The grids should load after both selectors have resolved the plans endpoint.

## Pre-merge Checklist
